### ISSUE TITLE
[Bug]: Model.getStringProperty$default crash due to NullPointerException

### ISSUE THREAD

**Issue by fanwgwg** [OneSignal/OneSignal-Android-SDK#1839]
### What happened?

Sometimes at app start, the app would randomly crash (saw 2 times out of ~50 attempts) at app open.

Crash:
```
Error attempting to execute operation: [com.onesignal.core.internal.operations.impl.OperationRepo$OperationQueueItem@24a7f5e]
                                                                                                    java.lang.NullPointerException: null cannot be cast to non-null type kotlin.String
                                                                                                    	at com.onesignal.common.modeling.Model.getStringProperty(Model.kt:215)
                                                                                                    	at com.onesignal.common.modeling.Model.getStringProperty$default(Model.kt:215)
                                                                                                    	at com.onesignal.common.modeling.Model.getId(Model.kt:61)
                                                                                                    	at com.onesignal.common.modeling.ModelStore.get(ModelStore.kt:63)
                                                                                                    	at com.onesignal.common.modeling.SingletonModelStore.getModel(SingletonModelStore.kt:22)
                                                                                                    	at com.onesignal.user.internal.operations.impl.executors.RefreshUserOperationExecutor.getUser(RefreshUserOperationExecutor.kt:110)
                                                                                                    	at com.onesignal.user.internal.operations.impl.executors.RefreshUserOperationExecutor.access$getUser(RefreshUserOperationExecutor.kt:27)
                                                                                                    	at com.onesignal.user.internal.operations.impl.executors.RefreshUserOperationExecutor$getUser$1.invokeSuspend(Unknown Source:15)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:33)
                                                                                                    	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                                                                                                    	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
                                                                                                    	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
                                                                                                    	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
                                                                                                    	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
                                                                                                    	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
                                                                                                    	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
                                                                                                    	at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$2.invoke(ThreadUtils.kt:74)
                                                                                                    	at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$2.invoke(ThreadUtils.kt:73)
                                                                                                    	at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
```

### Steps to reproduce?

```Markdown
Described above
```

### What did you expect to happen?

App should not crash

### OneSignal Android SDK version

5.0.1

### Android version

13

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
Hi @fanwgwg thank you for your patience here! We are still attempting to reproduce this crash. Is there any additional information you might be able to share about when you experience this? Appreciate it!

**Comment by dmytroreutov** [comment]
Hi, I'm observing same issue in production

**Comment by isles1217** [comment]
I'm also encountering the same crash, One Signal version 5.0.3. Stack trace:
```
Fatal Exception: java.lang.NullPointerException: null cannot be cast to non-null type kotlin.String
       at com.onesignal.common.modeling.Model.getStringProperty(Model.java:11)
       at com.onesignal.common.modeling.Model.getStringProperty$default(Model.java:7)
       at com.onesignal.common.modeling.Model.getId(Model.java:4)
       at com.onesignal.common.modeling.ModelStore.get(ModelStore.java:24)
       at com.onesignal.common.modeling.SingletonModelStore.getModel(SingletonModelStore.java:4)
       at com.onesignal.user.internal.operations.impl.listeners.SubscriptionModelStoreListener.getUpdateOperation(SubscriptionModelStoreListener.java:37)
       at com.onesignal.user.internal.operations.impl.listeners.SubscriptionModelStoreListener.getUpdateOperation(SubscriptionModelStoreListener.java:1)
       at com.onesignal.core.internal.operations.listeners.ModelStoreListener.onModelUpdated(ModelStoreListener.java:45)
       at com.onesignal.common.modeling.ModelStore$onChanged$1.invoke(ModelStore.java:2)
       at com.onesignal.common.modeling.ModelStore$onChanged$1.invoke(ModelStore.java:1)
       at com.onesignal.common.events.EventProducer.fire(EventProducer.java:21)
       at com.onesignal.common.modeling.ModelStore.onChanged(ModelStore.java:20)
       at com.onesignal.common.modeling.Model$notifyChanged$1.invoke(Model.java:2)
       at com.onesignal.common.modeling.Model$notifyChanged$1.invoke(Model.java:1)
       at com.onesignal.common.events.EventProducer.fire(EventProducer.java:21)
       at com.onesignal.common.modeling.Model.notifyChanged(Model.java:18)
       at com.onesignal.common.modeling.Model.setOptAnyProperty(Model.java:49)
       at com.onesignal.user.internal.subscriptions.SubscriptionModel.setStatus(SubscriptionModel.java:14)
       at com.onesignal.user.internal.subscriptions.impl.SubscriptionManager.addOrUpdatePushSubscription(SubscriptionManager.java:43)
       at com.onesignal.notifications.internal.listeners.DeviceRegistrationListener$retrievePushTokenAndUpdateSubscription$1.invokeSuspend(DeviceRegistrationListener.java:70)
       at com.onesignal.notifications.internal.listeners.DeviceRegistrationListener$retrievePushTokenAndUpdateSubscription$1.invoke(DeviceRegistrationListener.java:2)
       at com.onesignal.notifications.internal.listeners.DeviceRegistrationListener$retrievePushTokenAndUpdateSubscription$1.invoke(DeviceRegistrationListener.java:1)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1$1.invokeSuspend(ThreadUtils.kt:28)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(BaseContinuationImpl.java:8)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.java:105)
       at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoopImplBase.java:162)
       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(BlockingCoroutine.java:98)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(BuildersKt__Builders.kt:98)
       at kotlinx.coroutines.BuildersKt.runBlocking(Builders.kt:98)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(BuildersKt__Builders.kt:2)
       at kotlinx.coroutines.BuildersKt.runBlocking$default(Builders.kt:2)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:2)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:1)
       at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Threads.kt:2)
```

**Comment by brismithers** [comment]
Thank you all for your patience, unfortunately we are struggling to reproduce this on our end.  I suspect a race condition during initialization, but I can't identify it looking through the code.

If you are able to capture any logs beyond the stack trace (preferably with `OneSignal.getDebug().setLogLevel(LogLevel.VERBOSE);` set), that will help us understand the sequence of events.

It might also help to understand where `OneSignal.initWithContext` called, if that could be provided I'd appreciate it.

**Comment by isles1217** [comment]
Have you ruled out that this isn't a proguard/obfuscation issue? That's the smell I'm getting from this.

**Comment by isles1217** [comment]
If it's not a proguard/obfuscation problem, the simplest way to guard against initialization race conditions is to wrap relevant member variables in `lazy` initializers (assuming Kotlin here).

**Comment by brismithers** [comment]
We haven't ruled out proguard or obfuscation as the issue, but from the stack trace it looks tied to how we save and load state across app lifecycles.  @isles1217 if you could provide a more complete log showing the initialization prior to the stack trace, or provided more context in how OneSignal is initialized within your app, that would be tremendously helpful.  Thanks for the insights!

**Comment by isles1217** [comment]
@brismithers see this comment for initialization logic: https://github.com/OneSignal/OneSignal-Android-SDK/issues/1880#issuecomment-1787886484

**Comment by ipapps** [comment]
Happens also in the Flutter lib 5.0.4. This is causing crashes specifically on Android 13. Please give us some visibility on how you will tackle this problem as it is a critical bug !

**Comment by jkasten2** [comment]
I believe this is the same as issue #1995 and a fix for this is now available in [OneSignal-Android-SDK 5.1.6
](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.1.6)

**Comment by jkasten2** [comment]
@gajicm93 can you create a new issue with your full details? This issue is pretty old so the code around this has probably changed significantly.
